src/raft.rs

```diff
@@ -568,6 +568,17 @@ impl<T: Storage> Raft<T> {
         self.send(m);
     }
 
+    fn send_append_with_bcast_targets(&mut self, to: u64, pr: &mut Progress, targets: Vec<u64>) {
+        let old_msg_len = self.msgs.len();
+        self.send_append(to, pr);
+        let new_msg_len = self.msgs.len();
+        if new_msg_len > old_msg_len {
+            self.msgs[old_msg_len].set_bcast_targets(targets);
+        } else {
+            // TODO(qupeng): is it possible?
```

Comment:
I think it's impossible because we just receive the `MsgAppendResponse` from `to`